runtime/doc/pi_health.txt

```diff
@@ -35,15 +35,98 @@ Commands				*health-commands*
                         :checkhealth nvim
 <                      To run the healthchecks for the "foo" and "bar" plugins
                       (assuming these plugins are on your 'runtimepath' and
-                      they have implemented health#foo#check() and
-                      health#bar#check(), respectively): >
+                      they have implemented the Lua or Vimscript interface
+                      require("foo.health").check()
+                      and health#bar#check(), respectively): >
                         :checkhealth foo bar
 <
+                      To run healthcheck for lua submodules, for example nvim
+                      provides `vim.lsp`  or `vim.treesitter` it also accpets an
+                      "*" for all the submodules:  >
+                        :checkhealth vim.lsp vim.treesitter
+                        :checkhealth vim*
+<
+==============================================================================
+Functions				*health-functions-lua*
+
+health.vim lua functions are for creating new healthchecks. They mostly just
+do some layout and formatting, to give users a consistent presentation. See
+also |health-functions-vim|
+
+Lua					|health-lua|
+
+The lua health module provides the functions:
+
+  `local health = require('health')`
+
+health.report_start({name})				*health.report_start()*
+	Starts a new report. Most plugins should call this only once, but if
+        you want different sections to appear in your report, call this once
+        per section.
```

Comment:
Use hard tabs for indentation in help docs (the help file should have a modeline already with something like `noet ts=8 sw=0 tw=78 ft=help`, feel free to add it if it doesn't exist)

Reply:
I went ahead and changed all the indentation for tabs! thanks for the review.